rust/gitxetcore/src/git_integration/git_repo.rs

```diff
@@ -308,10 +372,25 @@ impl GitRepo {
             // we explicitly have at this point.  The filter will run on clone before the remote notes
             // have been fetched, so in this case we need to explicitly fetch them.
             let remotes = Self::list_remote_names(s.repo.clone())?;
-            for r in remotes {
-                s.sync_remote_to_notes(&r)?;
+
+            for r in remotes.iter() {
+                s.sync_remote_to_notes(r)?;
+            }
+
+            if let Some(upstream_repo) = &s.xet_config.upstream_xet_repo {
+                info!("GitRepo::open: Adding remote upstream url {upstream_repo:?} to remotes to scan.");
+
+                // Create the remote url using the example form of the origin repo.
+                for remote_url in s.get_xet_upstream_remote_urls(upstream_repo)? {
+                    let _ = s.sync_remote_to_notes_with_custom_destination(
+                    &remote_url,
+                    Some("_upstream_xet_repo_"),
```

Comment:
Where is this alternative remote set?

Reply:
So the backup, if that .xet/config.toml file isn't set, is to have the users manually add a remote that we can pull the repo salt from. 

The _upstream_xet_repo_ is just the location where the refs from that repo will be dumped on fetch.  It's not an actually remote that set.

Reply:
I see.